ssntp/frame.go

```diff
@@ -291,3 +292,14 @@ func (f Frame) DumpTrace() (*payloads.FrameTrace, error) {
 
 	return &s, nil
 }
+
+// OriginUUID is the frame first sender and creator UUID.
+// When a SSNTP frame is forwarded by a server, the client
+// then only sees a new frame coming but it can not tell
+// who the frame creator and first sender is. This method
+// allows to fetch such information from a frame.
+func (f Frame) OriginUUID() (origin uuid.UUID) {
+	copy(origin[:], f.Origin[:16])
```

Comment:
@markdryan so you're suggesting that I check if len(f.Origin) == 16 ?
It's in theory not possible for a SSNTP client to craft a frame with an Origin UUID of less than 16 bytes.